prow/apis/prowjobs/v1/types.go

```diff
@@ -969,6 +975,9 @@ func (g *GCSConfiguration) ApplyDefault(def *GCSConfiguration) *GCSConfiguration
 	if merged.LocalOutputDir == "" {
 		merged.LocalOutputDir = def.LocalOutputDir
 	}
+	if len(merged.CompressFileTypes) == 0 {
+		merged.CompressFileTypes = def.CompressFileTypes
+	}
```

Comment:
This doesn't provide any way for a more specific config to opt-out if a less specific config opts-in. For example, if CompressFileTypes are configured for a repo, but a job explicitly specified an empty `compress_file_types` field, the job will still inherit the value configured for the repo.

I think a `nil` check would be more appropriate here. We should also have a [unit test](https://github.com/kubernetes/test-infra/blob/eee69be89fe8cd51b14909ac97484a8d86cfa17f/prow/apis/prowjobs/v1/types_test.go#L169) to validate this defaulting behavior.

Reply:
That's a good point. I will make that update.